### DISCUSSION TITLE
serviceAccount certificate

### DISCUSSION THREAD

**Discussion by ilvpozzd** [siderolabs/talos#5338] [Q&A] [answered]
What serviceAccount certificate is being used by default?

https://www.talos.dev/v1.0/reference/configuration/#clusterconfig - serviceAccount block

**Comment by smira** [comment, marked as answer]
It's a key, and a random one is generated whenever Talos machine configuration is generated for each cluster.

**Reply by ilvpozzd** [reply]
There is a way to specify certificate owned by me?

**Reply by smira** [reply]
you can modify generated machine configuration to suit your needs freely

**Reply by ilvpozzd** [reply]
What algorithm is used to generate the certificate? The final case is for me to know the data of the standard admin SA (by generating from my side and passing to Talos, or by receiving from Talos or retrying) for further authentication in the Kubernetes cluster

**Reply by frezbo** [reply]
https://www.siderolabs.com/blog/workload-identity-for-kubernetes-on-gcp/ the blog has an example on manually passing a SA key, by default talos uses ecdsa keys

**Reply by frezbo** [reply]
kube-apiserver supports both RSA and ECDSA keys

**Reply by ilvpozzd** [reply]
Well thank you. And the last question - is the serviceAccount key used to create the "default-admin" SA or something else?

**Reply by frezbo** [reply]
it's the signing key, the private key is used to sign SA token when using projected volume mounts
